### PR TITLE
This PR fixes the node startup issue - https://github.com/corda/corda/issues/37

### PR DESCRIPTION
**Pull request by patrickkuo**
*Create missing artemis bridge when node is added to the network map
*Added integration test for this issue

### CODE REVIEW

File: node/src/main/kotlin/net/corda/node/services/messaging/ArtemisMessagingServer.kt [hunk 1]
```diff
@@ -108,13 +108,14 @@ class ArtemisMessagingServer(override val config: NodeConfiguration,
         maybeDeployBridgeForAddress(networkMapService)
     }
 
-    private fun destroyPossibleStaleBridge(change: MapChange) {
-        val staleNodeInfo = when (change) {
-            is MapChange.Modified -> change.previousNode
-            is MapChange.Removed -> change.node
-            is MapChange.Added -> return
+    private fun destroyOrCreateBridge(change: MapChange) {
```
Comment (shamsasari) [inline node/src/main/kotlin/net/corda/node/services/messaging/ArtemisMessagingServer.kt]: Oh dear my bad! - Can you add doc for why this method needs to add bridges even though the queue creation handler does it already. This is what got me.

File: node/src/test/kotlin/net/corda/node/services/messaging/ArtemisMessagingServerTest.kt [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+import net.corda.node.driver.driver
+import org.junit.Test
+import java.nio.file.Paths
+import java.time.Instant
+import java.time.ZoneOffset
+import java.time.format.DateTimeFormatter
+import java.util.concurrent.TimeUnit
+
+class ArtemisMessagingServerTest {
```
Comment (shamsasari) [inline node/src/test/kotlin/net/corda/node/services/messaging/ArtemisMessagingServerTest.kt]: Can you move this to `integration-test`

File: node/src/test/kotlin/net/corda/node/services/messaging/ArtemisMessagingServerTest.kt [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    @Test
+    fun `test network map will work after restart`() {
```
Comment (shamsasari) [inline node/src/test/kotlin/net/corda/node/services/messaging/ArtemisMessagingServerTest.kt]: "test" is a bit redundant

**Comment by mikehearn** [comment]
This fix is still a bit indirect. Now what happens if we stop issuing network map change observations at startup e.g. due to loading from a local cache - the same issue will return.

Why not just do things directly: at startup, iterate over the queues that exist and then call `deployBridgeFromNewPeerQueue` on them? It can be renamed to `deployBridgeFromPeerQueue` at the same time. That way it's obvious what's happening during startup and you aren't relying on indirect effects to trigger it (which when relied upon definitely need a big comment explaining so!)

**Comment by patrickkuo** [comment]
Hi @mikehearn. Creating the bridge will need the hostAndPort from the network map. However the network map is empty on startup as we are not persisting it (i think).
Persisting network map might take an extra few days to code and tested. Should i do it in this change or create another PR for that?

**Comment by mikehearn** [comment]
Ah, I see. OK, for now just add a comment above the line in start() saying what assumptions you're making here in order to construct bridges.

**Pull request merged**